### PR TITLE
add sequelize section to Dev Guide 3rd party tools compatibility

### PR DESCRIPTION
**Pull request by shczhen**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [ ] master (the latest development version)
- [ ] v6.4 (TiDB 6.4 versions)
- [ ] v6.3 (TiDB 6.3 versions)
- [x] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: develop/dev-guide-third-party-tools-compatibility.md [hunk 1]
```diff
@@ -156,3 +156,75 @@ TiDB 对其进行了两个维度的修复：
...
+
+- 不支持与外键约束相关的功能(包括多对多关联)；
+- 不支持`GEOMETRY`相关；
+- 不支持删除整数主键；
+- 不支持`PROCEDURE`相关；
+- 不支持`SERIALIZABLE`隔离级别；
+- 默认不允许修改列的`AUTO_INCREMENT`属性；
+- 不支持`FULLTEXT`、`HASH`和`SPATIAL`索引；
+
+### 不支持与外键约束相关的功能
```
Comment (Icemap) [inline develop/dev-guide-third-party-tools-compatibility.md:174]: 外键相关不支持已于本页面开始提示，无需重复。

File: develop/dev-guide-third-party-tools-compatibility.md [hunk 2]
```diff
@@ -156,3 +156,75 @@ TiDB 对其进行了两个维度的修复：
...
+
+**描述**
+
+1. TiDB 不支持外键约束。
+
+2. [Sequelize 的多对多关系](https://sequelize.org/docs/v6/core-concepts/assocs/#philosophy-2)使用了一个额外的模型，该模型将具有两个外键列。因此，Sequelize 无法在 TiDB 上使用多对多关系。
+
+### 不支持`GEOMETRY`相关
```
Comment (Icemap) [inline develop/dev-guide-third-party-tools-compatibility.md:182]: ditto

File: develop/dev-guide-third-party-tools-compatibility.md [hunk 3]
```diff
@@ -156,3 +156,75 @@ TiDB 对其进行了两个维度的修复：
...
+
+1. 不支持删除整数类型的主键 [#18090](https://github.com/pingcap/tidb/issues/18090)；
+
+2. 当表的 `pkIsHandle` 变量为 `true` 时，不支持删除主键；
+
+**规避方法**
+
+[使用 `AUTO_RANDOM` 处理自增主键热点表](/troubleshoot-hot-spot-issues.md#使用-auto_random-处理自增主键热点表)，而不是 `AUTO_INCREMENT`。
+
+### 不支持`PROCEDURE`相关
```
Comment (Icemap) [inline develop/dev-guide-third-party-tools-compatibility.md:200]: ditto

File: develop/dev-guide-third-party-tools-compatibility.md [hunk 4]
```diff
@@ -156,3 +156,75 @@ TiDB 对其进行了两个维度的修复：
...
+- 不支持`SERIALIZABLE`隔离级别；
+- 默认不允许修改列的`AUTO_INCREMENT`属性；
+- 不支持`FULLTEXT`、`HASH`和`SPATIAL`索引；
+
+### 不支持与外键约束相关的功能
+
+**描述**
+
+1. TiDB 不支持外键约束。
+
```
Comment (Icemap) [inline develop/dev-guide-third-party-tools-compatibility.md:179]:
```suggestion
```

File: develop/dev-guide-third-party-tools-compatibility.md [hunk 5]
```diff
@@ -156,3 +156,75 @@ TiDB 对其进行了两个维度的修复：
...
+**描述**
+
+TiDB 不支持空间类型的函数（即`SPATIAL`，或者`GIS`/`GEOMETRY`）、数据类型和索引。
+
+### 不支持删除整数主键
+
+**描述**
+
+1. 不支持删除整数类型的主键 [#18090](https://github.com/pingcap/tidb/issues/18090)；
+
```
Comment (Icemap) [inline develop/dev-guide-third-party-tools-compatibility.md:193]:
```suggestion
```

File: develop/dev-guide-third-party-tools-compatibility.md [hunk 6]
```diff
@@ -156,3 +156,54 @@ TiDB 对其进行了两个维度的修复：
...
+### 不支持删除整数主键
+
+**描述**
+
+1. 不支持删除整数类型的主键 [#18090](https://github.com/pingcap/tidb/issues/18090)；
+2. 当表的 `pkIsHandle` 变量为 `true` 时，不支持删除主键；
+
+**规避方法**
+
+[使用 `AUTO_RANDOM` 处理自增主键热点表](/troubleshoot-hot-spot-issues.md#使用-auto_random-处理自增主键热点表)，而不是 `AUTO_INCREMENT`。
```
Comment (Icemap) [inline develop/dev-guide-third-party-tools-compatibility.md:183]:
```suggestion
### 不支持修改整数主键

**描述**

不支持修改整数类型的主键，这是由于当主键为整数类型时，TiDB 使用其作为数据组织的索引。你可以在此 [Issue](https://github.com/pingcap/tidb/issues/18090) 或[聚簇索引](/clustered-indexes.md)一节中获取更多信息。
```

File: develop/dev-guide-third-party-tools-compatibility.md [hunk 7]
```diff
@@ -156,3 +156,54 @@ TiDB 对其进行了两个维度的修复：
...
+
+## 与 Sequelize 的兼容性
+
+基于 [Sequelize v6.21.4](https://www.npmjs.com/package/sequelize/v/6.21.4) 测试。
+
+根据测试结果，TiDB 支持绝大部分 Sequelize 功能（使用 MySQL8 作为方言），不支持的功能有：
+
+- 不支持与外键约束相关的功能(包括多对多关联)；
+- 不支持`GEOMETRY`相关；
+- 不支持删除整数主键；
```
Comment (Icemap) [inline develop/dev-guide-third-party-tools-compatibility.md:168]:
```suggestion
- 不支持修改整数主键；
```

File: develop/dev-guide-third-party-tools-compatibility.md [hunk 8]
```diff
@@ -156,3 +156,58 @@ TiDB 对其进行了两个维度的修复：
...
+
+**描述**
+
+TiDB 不支持`SERIALIZABLE`隔离级别。
+
+**规避方法**
+
+使用 TiDB 支持的 `REPEATABLE-READ` 或 `READ-COMMITTED` 隔离级别。
+
+如果目的是兼容其他设置（但不依赖） `SERIALIZABLE` 隔离级别的应用，可以设置 `tidb_skip_isolation_level_check` 为 `1`，此后如果对 `tx_isolation`（`transaction_isolation` 别名）赋值一个 TiDB 不支持的隔离级别（`READ-UNCOMMITTED` 和 `SERIALIZABLE`），不会报错。
```
Comment (Icemap) [inline develop/dev-guide-third-party-tools-compatibility.md:190]:
```suggestion
### 不支持 `READ-UNCOMMITTED`，`SERIALIZABLE` 隔离级别

**描述**

TiDB 不支持 `READ-UNCOMMITTED`，`SERIALIZABLE` 隔离级别。设置事物隔离级别为 `READ-UNCOMMITTED`，`SERIALIZABLE` 时将报错。

**规避方法**

仅使用 TiDB 支持的 `REPEATABLE-READ` 或 `READ-COMMITTED` 隔离级别。

如果目的是兼容其他设置（但不依赖） `SERIALIZABLE` 隔离级别的应用，可以设置 [`tidb_skip_isolation_level_check`](/system-variables.md#tidb_skip_isolation_level_check) 为 `1`，此后如果对 `tx_isolation`（`transaction_isolation` 别名）赋值一个 TiDB 不支持的隔离级别（`READ-UNCOMMITTED` 和 `SERIALIZABLE`），不会报错。
```

File: develop/dev-guide-third-party-tools-compatibility.md [hunk 9]
```diff
@@ -156,3 +156,53 @@ TiDB 对其进行了两个维度的修复：
...
+基于 [Sequelize v6.21.4](https://www.npmjs.com/package/sequelize/v/6.21.4) 测试。
+
+根据测试结果，TiDB 支持绝大部分 Sequelize 功能（使用 MySQL8 作为方言），不支持的功能有：
+
+- 不支持与外键约束相关的功能(包括多对多关联)；
+- 不支持`GEOMETRY`相关；
+- 不支持修改整数主键；
+- 不支持`PROCEDURE`相关；
+- 不支持`SERIALIZABLE`隔离级别；
+- 默认不允许修改列的`AUTO_INCREMENT`属性；
```
Comment (Icemap) [inline develop/dev-guide-third-party-tools-compatibility.md:171]:
```suggestion
- [不支持 `GEOMETRY`](https://github.com/pingcap/tidb/issues/6347) 相关；
- 不支持修改整数主键；
- 不支持 `PROCEDURE` 相关；
- 不支持 `READ-UNCOMMITTED`，`SERIALIZABLE` [隔离级别](/system-variables.md#transaction_isolation)；
- 默认不允许修改列的 `AUTO_INCREMENT` 属性；
```

File: develop/dev-guide-third-party-tools-compatibility.md [hunk 10]
```diff
@@ -156,3 +156,53 @@ TiDB 对其进行了两个维度的修复：
 
 - 客户端方面：**pingcap/mysql-connector-j** 中修复了该 Bug，你可以使用 [pingcap/mysql-connector-j](https://github.com/pingcap/mysql-connector-j) 替换官方的 MySQL Connector/J。
 - 服务端方面：TiDB v6.3.0 修复了此兼容性问题，你可以升级服务端至 v6.3.0 或以上版本。
+
+## 与 Sequelize 的兼容性
+
+基于 [Sequelize v6.21.4](https://www.npmjs.com/package/sequelize/v/6.21.4) 测试。
+
+根据测试结果，TiDB 支持绝大部分 Sequelize 功能（使用 MySQL8 作为方言），不支持的功能有：
```
Comment (ran-huang) [inline develop/dev-guide-third-party-tools-compatibility.md:164]: MySQL8 dialect 能否贴个链接

File: develop/dev-guide-third-party-tools-compatibility.md [hunk 11]
```diff
@@ -156,3 +156,53 @@ TiDB 对其进行了两个维度的修复：
...
+
+- 不支持与外键约束相关的功能(包括多对多关联)；
+- [不支持 `GEOMETRY`](https://github.com/pingcap/tidb/issues/6347) 相关；
+- 不支持修改整数主键；
+- 不支持 `PROCEDURE` 相关；
+- 不支持 `READ-UNCOMMITTED`，`SERIALIZABLE` [隔离级别](/system-variables.md#transaction_isolation)；
+- 默认不允许修改列的 `AUTO_INCREMENT` 属性；
+- 不支持`FULLTEXT`、`HASH`和`SPATIAL`索引；
```
Comment (ran-huang) [inline develop/dev-guide-third-party-tools-compatibility.md:172]:
```suggestion
- 不支持与外键约束相关的功能（包括多对多关联）。
- [不支持 `GEOMETRY`](https://github.com/pingcap/tidb/issues/6347) 相关。
- 不支持修改整数主键。
- 不支持 `PROCEDURE` 相关。
- 不支持 `READ-UNCOMMITTED` 和 `SERIALIZABLE` [隔离级别](/system-variables.md#transaction_isolation)。
- 默认不允许修改列的 `AUTO_INCREMENT` 属性。
- 不支持 `FULLTEXT`、`HASH` 和 `SPATIAL` 索引。
```

File: develop/dev-guide-third-party-tools-compatibility.md [hunk 12]
```diff
@@ -156,3 +156,53 @@ TiDB 对其进行了两个维度的修复：
...
+
+### 不支持修改整数主键
+
+**描述**
+
+不支持修改整数类型的主键，这是由于当主键为整数类型时，TiDB 使用其作为数据组织的索引。你可以在此 [Issue](https://github.com/pingcap/tidb/issues/18090) 或[聚簇索引](/clustered-indexes.md)一节中获取更多信息。
+
+### 不支持 `READ-UNCOMMITTED`，`SERIALIZABLE` 隔离级别
```
Comment (ran-huang) [inline develop/dev-guide-third-party-tools-compatibility.md:180]:
```suggestion
### 不支持 `READ-UNCOMMITTED` 和 `SERIALIZABLE` 隔离级别
```

File: develop/dev-guide-third-party-tools-compatibility.md [hunk 13]
```diff
@@ -156,3 +156,53 @@ TiDB 对其进行了两个维度的修复：
...
+
+**描述**
+
+TiDB 不支持 `READ-UNCOMMITTED`，`SERIALIZABLE` 隔离级别。设置事务隔离级别为 `READ-UNCOMMITTED`，`SERIALIZABLE` 时将报错。
```
Comment (ran-huang) [inline develop/dev-guide-third-party-tools-compatibility.md:184]:
```suggestion
TiDB 不支持 `READ-UNCOMMITTED` 和 `SERIALIZABLE` 隔离级别。设置事务隔离级别为 `READ-UNCOMMITTED` 或 `SERIALIZABLE` 时将报错。
```

File: develop/dev-guide-third-party-tools-compatibility.md [hunk 14]
```diff
@@ -156,3 +156,53 @@ TiDB 对其进行了两个维度的修复：
...
+
+**规避方法**
+
+仅使用 TiDB 支持的 `REPEATABLE-READ` 或 `READ-COMMITTED` 隔离级别。
+
+如果目的是兼容其他设置（但不依赖） `SERIALIZABLE` 隔离级别的应用，可以设置 [`tidb_skip_isolation_level_check`](/system-variables.md#tidb_skip_isolation_level_check) 为 `1`，此后如果对 `tx_isolation`（`transaction_isolation` 别名）赋值一个 TiDB 不支持的隔离级别（`READ-UNCOMMITTED` 和 `SERIALIZABLE`），不会报错。
```
Comment (ran-huang) [inline develop/dev-guide-third-party-tools-compatibility.md:190]: 这里使用 `tx_isolation` 有什么特别的含义吗？没有的话直接用 `transaction_isolation` 吧
```suggestion
如果你的目的是兼容其他设置 `SERIALIZABLE` 隔离级别的应用，但不依赖于 `SERIALIZABLE`，你可以设置 [`tidb_skip_isolation_level_check`](/system-variables.md#tidb_skip_isolation_level_check) 为 `1`，此后如果对 `tx_isolation`（[`transaction_isolation`](/system-variable.md#transaction_isolation) 别名）赋值一个 TiDB 不支持的隔离级别（`READ-UNCOMMITTED` 和 `SERIALIZABLE`），不会报错。
```

File: develop/dev-guide-third-party-tools-compatibility.md [hunk 15]
```diff
@@ -156,3 +156,53 @@ TiDB 对其进行了两个维度的修复：
...
+
+### 默认不允许修改列的`AUTO_INCREMENT`属性
```
Comment (ran-huang) [inline develop/dev-guide-third-party-tools-compatibility.md:192]:
```suggestion
### 默认不允许修改列的 `AUTO_INCREMENT` 属性
```

File: develop/dev-guide-third-party-tools-compatibility.md [hunk 16]
```diff
@@ -156,3 +156,53 @@ TiDB 对其进行了两个维度的修复：
...
+
+参考 [`AUTO_INCREMENT` 的使用限制](/auto-increment.md#使用限制)。
+
+设置 `@@tidb_allow_remove_auto_inc` 为 `true`，即可允许移除 `AUTO_INCREMENT` 属性。
+
+### 不支持`FULLTEXT`、`HASH`和`SPATIAL`索引
+
+**描述**
+
+TiDB 不支持`FULLTEXT`、`HASH`和`SPATIAL`索引。
```
Comment (ran-huang) [inline develop/dev-guide-third-party-tools-compatibility.md:208]:
```suggestion
TiDB 不支持 `FULLTEXT`、`HASH` 和 `SPATIAL` 索引。
```

File: develop/dev-guide-third-party-tools-compatibility.md [hunk 17]
```diff
@@ -156,3 +156,53 @@ TiDB 对其进行了两个维度的修复：
...
+
+默认不允许通过 `ALTER TABLE MODIFY` 或 `ALTER TABLE CHANGE` 来增加或移除某个列的 `AUTO_INCREMENT` 属性。
+
+**规避方法**
+
+参考 [`AUTO_INCREMENT` 的使用限制](/auto-increment.md#使用限制)。
+
+设置 `@@tidb_allow_remove_auto_inc` 为 `true`，即可允许移除 `AUTO_INCREMENT` 属性。
+
+### 不支持`FULLTEXT`、`HASH`和`SPATIAL`索引
```
Comment (ran-huang) [inline develop/dev-guide-third-party-tools-compatibility.md:204]:
```suggestion
### 不支持 `FULLTEXT`、`HASH` 和 `SPATIAL` 索引
```

**Review by ran-huang — COMMENTED** [review]
Minor comments. Rest LGTM

File: develop/dev-guide-third-party-tools-compatibility.md (hunk 10 above)
Comment (shczhen) [reply develop/dev-guide-third-party-tools-compatibility.md:164]: https://sequelize.org/docs/v6/other-topics/dialect-specific-things/#mysql
那就用这个链接如何。
此外我会把`MySQL8`改成`MySQL`，避免sequelize的`MySQL2` driver 和 mysql的`MySQL8` 版本概念混淆。

Comment (ran-huang) [reply develop/dev-guide-third-party-tools-compatibility.md:164]: lgtm

**Review by ran-huang — APPROVED** [review]

**Review by Icemap — APPROVED** [review]

**Review by ran-huang — APPROVED** [review]

**Pull request merged**